ipatests/test_integration/test_idp.py

```diff
@@ -97,20 +102,199 @@ def install(cls, mh):
+
+    def test_idp_with_services(self):
+        """
+        Test case to check that services can be configured
+        auth indicator as idp.
+        """
         tasks.clear_sssd_cache(self.master)
-        kinit_idp(self.master, 'keycloakuser', keycloak_srv)
+        tasks.kinit_admin(self.master)
+        domain = self.master.domain.name.upper()
+        services = [
+            "DNS/{0}@{1}".format(self.master.hostname, domain),
+            "HTTP/{0}@{1}".format(self.client.hostname, domain),
+            "dogtag/{0}@{1}".format(self.master.hostname, domain),
+            "ipa-dnskeysyncd/{0}@{1}".format(self.master.hostname, domain)
+        ]
+        try:
+            for service in services:
+                test = self.master.run_command(["ipa", "service-mod", service,
+                                                "--auth-ind=idp"]
+                                               )
+                assert "Authentication Indicators: idp" in test.stdout_text
+        finally:
+            for service in services:
+                self.master.run_command(["ipa", "service-mod", service,
+                                         "--auth-ind="])
+
+    def test_idp_backup_restore(self):
+        """
+        Test case to check that after restore data is retrieved
+        with related idp configuration.
+        """
+        tasks.kinit_admin(self.master)
+        user = "backupuser"
+        cmd = ["ipa", "idp-add", "testidp", "--provider=keycloak",
+               "--client-id=ipa_oidc_client", "--org=master",
+               "--base-url={0}:8443/auth".format(self.client.hostname)]
+        self.master.run_command(cmd, stdin_text="{0}\n{0}".format(
+            self.client.config.admin_password))
+
+        tasks.user_add(self.master, user,
+                       extra_args=["--user-auth-type=idp",
+                                   "--idp-user-id=testuser1@ipa.test",
+                                   "--idp=testidp"]
+                       )
+
+        backup_path = tasks.get_backup_dir(self.master)
+        # change data after backup
+        self.master.run_command(['ipa', 'user-del', user])
+        self.master.run_command(['ipa', 'idp-del', 'testidp'])
+        dirman_password = self.master.config.dirman_password
+        self.master.run_command(['ipa-restore', backup_path],
+                                stdin_text=dirman_password + '\nyes')
+        try:
+            list_user = self.master.run_command(
+                ['ipa', 'user-show', 'backupuser', '--all']
+            ).stdout_text
+            assert "External IdP configuration: testidp" in list_user
+            assert "User authentication types: idp" in list_user
+            assert ("External IdP user identifier: "
+                    "testuser1@ipa.test") in list_user
+            list_idp = self.master.run_command(['ipa', 'idp-find', 'testidp'])
+            assert "testidp" in list_idp.stdout_text
```

Comment:
I would also check that authentication with external IdP is also working after the restore.